images/virtualization-artifact/pkg/controller/vd/internal/resizing.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2024 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package internal
+
+import (
+	"context"
+	"errors"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/deckhouse/virtualization-controller/pkg/controller/common"
+	"github.com/deckhouse/virtualization-controller/pkg/controller/service"
+	"github.com/deckhouse/virtualization-controller/pkg/controller/supplements"
+	virtv2 "github.com/deckhouse/virtualization/api/core/v1alpha2"
+	"github.com/deckhouse/virtualization/api/core/v1alpha2/vdcondition"
+)
+
+type ResizingHandler struct {
+	diskService *service.DiskService
+}
+
+func NewResizingHandler(diskService *service.DiskService) *ResizingHandler {
+	return &ResizingHandler{
+		diskService: diskService,
+	}
+}
+
+func (h ResizingHandler) Handle(ctx context.Context, vd *virtv2.VirtualDisk) (reconcile.Result, error) {
+	condition, ok := service.GetCondition(vdcondition.ResizedType, vd.Status.Conditions)
+	if !ok {
+		condition = metav1.Condition{
+			Type:   vdcondition.ResizedType,
+			Status: metav1.ConditionUnknown,
+		}
+	}
+
+	defer func() { service.SetCondition(condition, &vd.Status.Conditions) }()
+
+	if vd.DeletionTimestamp != nil {
+		condition.Status = metav1.ConditionUnknown
+		condition.Reason = ""
+		condition.Message = ""
+		return reconcile.Result{}, nil
+	}
+
+	newSize := vd.Spec.PersistentVolumeClaim.Size
+	if newSize == nil {
+		condition.Status = metav1.ConditionFalse
+		condition.Reason = vdcondition.ResizedReason_NotRequested
+		condition.Message = ""
+		return reconcile.Result{}, nil
+	}
+
+	readyCondition, ok := service.GetCondition(vdcondition.ReadyType, vd.Status.Conditions)
+	if !ok || readyCondition.Status != metav1.ConditionTrue {
+		condition.Status = metav1.ConditionFalse
+		condition.Reason = vdcondition.ResizedReason_NotRequested
+		condition.Message = ""
```

Comment:
```suggestion
		condition.Status = metav1.ConditionUnknown
		condition.Reason = ""
		condition.Message = ""
```
What do you think?
"NotRequested" gives me a clue that I can request to resize disk even if it is "not ready". But code does the opposite: resize is not possible until disk is Ready.

Reply:
I'm also in favor of using "Unknown," but we're waiting for a general approach to be developed; this topic has already been discussed.